### DISCUSSION TITLE
Measuring ping latency?

### DISCUSSION THREAD

**Discussion by colinmollenhour** [socketio/socket.io#4737] [Q&A]
Is there any way with the current version to measure the latency of the heartbeats? I'd like to capture some metrics but without sending extra packets if possible. Thanks in advance!

**Comment by darrachequesne** [comment]
Hi! I'm afraid this is not possible with the current version, since it's the server that starts the round trip by sending the PING packets.

**Comment by colinmollenhour** [comment]
Actually I was wanting to capture it at the server. So for example it woudl be great to do this:

```
socket.on('pong', (meta) => {
  console.log(`received ping response in ${meta.response_time} seconds.`)
})
```

Or are you saying the PING is a lower-level thing that is not even exposed via Node?
